.github/workflows/build.yml

```diff
@@ -137,8 +134,49 @@ jobs:
         with:
           path: test.oci
           key: test_container:${{ matrix.arch }}-${{ github.run_id }}
+  matrix:
+    name: generate matrices
+    runs-on: 'ubuntu-latest'
+    defaults:
+      run:
+        shell: bash
+    outputs:
+      matrix: ${{ steps.matrix.outputs.matrix }}
+      matrix_bare: ${{ steps.matrix_bare.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # pin@v4.1.1
+      - id: matrix
+        run: |
+          FLAVORS_AMD64=$(./bin/flavors_parse.py --exclude "bare-*" --no-arch --json-by-arch --build --test | jq -r '.amd64')
+          FLAVORS_ARM64=$(./bin/flavors_parse.py --exclude "bare-*" --no-arch --json-by-arch --build --test | jq -r '.arm64')
+          MATRIX=$(jq -nc \
+            --argjson amd64 "$FLAVORS_AMD64" \
+            --argjson arm64 "$FLAVORS_ARM64" \
+            '{
+              include: (
```

Comment:
why make a matrix made up entirely of includes only? couldn't we have a matrix of the form

```
arch: [ amd64, arm64 ]
flavors: [ ... ]
```

Where only the flavours get filled in automatically, but we don't need an explicit include entry for both of the flavour - arch pair

Reply:
We have scenarios where we do not build/publish for `arm64`, e.g. ali, so this has to be flexible.
We could work with excludes in this case (as we do now) but this is "working with exceptions" again and my approach was more like "name what you want to build and not what you don't want to build".